### DISCUSSION TITLE
[A-L] (2023/24) Foglio 9 - Esercizio 1

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/calcolo-delle-probabilita#192] [Esercizi - Bertini] [answered]
![Screenshot 2023-11-27 134854](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/107944476/9914cc3c-5333-450a-a383-dc2408cb8ae8)

**Comment by IonutCicio** [comment]
#### Calcolare la distribuzione di $X_1$

Prima di tutto è utile definire $\Omega$

$$
\Omega = \\{ (\omega_1, \omega_2, ..., \omega_k) \mid \omega_1 + \omega_2 + \cdots + \omega_k = n \\}
$$

Per calcolarne la cardinalità basta rifarsi al problema dei cerchi e delle barre ($\circ, \mid$), con $n$ cerchi (le palline da distribuire) e $k - 1$ barre per indicare i $k$ cassetti

$$
\circ_1 \circ_2 \cdots \circ_n \\ \mid_1 \\ \mid_2 \\ \cdots \\ \mid_{k - 1}
$$

Per cui $| \Omega | = {n + k - 1 \choose k - 1}$

Ora possiamo fare qualche caso base prima di procedere a generalizzare la distribuzione di $X_1$. Per $X_1 = 0$ basta fissare la **prima barra**, e scegliere le posizioni per le altre $k - 2$ fra le $n + k - 2$ posizioni rimanenti, visto che non c'è nessuna pallina nella prima scatola.

$$
\mathbb{P}(X_1 = 0) = \frac{ n + k - 2 \choose k - 2} { n + k - 1 \choose k - 1}
$$

Per $X_1 = 1$ basta fissare una pallina e la prima barra e scegliere le posizioni per le altre $k - 2$ barre restanti (ora ho $n + k - 2 - 1 = n + k - 3$ posizioni fra cui scegliere, dato che ho fissato 1 barra e 1 pallina)

$$
\mathbb{P}(X_1 = 1) = \frac{ n + k - 3 \choose k - 2} { n + k - 1 \choose k - 1}
$$

Più in generale, per $x = 0, 1, 2, ..., n$

$$
\mathbb{P}(X_1 = x) = \frac{ n + k - 2 - x \choose k - 2} { n + k - 1 \choose k - 1}
$$

> [!NOTE]
> Rispettivamente la distribuzione di $X_1$ per 100 palline e 6 scatole, e la distribuzione per 100 palline e 50 scatole
<img src="https://github.com/sapienzastudentsnetwork/calcprob2324/assets/40034429/4fea4ea5-8b46-4969-ab41-97b204262def" width=350 alt="100 palline 6 scatole">

<img src="https://github.com/sapienzastudentsnetwork/calcprob2324/assets/40034429/b154c42f-9cc4-4c83-8548-547c1a65f496" width=350 alt="100 palline 50 scatole">

> [!NOTE]
> Può succedere che le palline siano meno delle scatole. Questa è la distribuzione di $X_1$ per 98 palline e 200 scatole

<img src="https://github.com/sapienzastudentsnetwork/calcprob2324/assets/40034429/c4cc47dd-2870-40ea-a47a-342fd0f07863" width=350 alt="98 palline 200 scatole">

Di seguito il codice Python che ho usato per generare i grafici della distribuzione

```python
import matplotlib.pyplot as plt
from math import factorial
from random import randrange


def binomial(n, k):
    return factorial(n) / (factorial(n - k) * factorial(k))


def balls_distribution(n, k, x):
    return binomial(n + k - 2 - x, k - 2) / binomial(n + k - 1, k - 1)


cases = (
    (100, 4), (100, 6), (100, 50), (98, 200),
    (randrange(1, 500), randrange(1, 500)),
)


for n, k in cases:
    distribution = [0] * n
    for x in range(n):
        distribution[x] = balls_distribution(n, k, x)

    fig, ax = plt.subplots()
    data = ax.stairs(distribution, linewidth=(1 + 1/4) ** 2)
    plt.savefig(f'boxes-{n}-{k}.png', transparent=False)
```

**Comment by cornflexxx** [comment, marked as answer]
### Distribuzione di $X_1$
Provo a considerare in modo diverso la distribuzione delle palline nella scatola $1$, infatti una singola pallina può stare all'interno della scatola con probabilità $\frac{1}{k}$ , al contrario può cadere in una delle altre con probabilità $\frac{k-1}{k}$. Quindi:
$$X_1\sim Bin(\frac{1}{k}, n)$$

------------------------
### Calcolo del valore atteso di $X_1$
Per trovare la covarianza di $X_1$ e $X_2$ è necessario  calcolare il valore atteso della singola variabile aleatoria.

Considero la $v.a$ $X_{1i}$ che  vale 1 se la $i$ -esima pallina è nella scatola $1$, quindi:

$$X_1 = \sum_{i=1}^{n} X_{1i}$$

$$\mathbb{P}(X_{1i}=1)=\frac{1}{k}$$

Allora possiamo  sfruttare le  proprietà del valore atteso per calcolarlo in modo semplice :

$$\mathbb{E}(X_1)=\mathbb{E}(\sum_{i=1}^{n} X_{1i})=\sum_{i=1}^{n}\mathbb{E}(X_{1i})$$

Calcolo il valore atteso di $X_{1i}$:

$$\mathbb{E}(X_{1i})=\frac{1}{k}$$

$$\sum_{i=1}^{n}\mathbb{E}(X_{1i})=\frac{n}{k}$$

(In realtà dato che $X_1$ ha distribuzione binomiale potevo omettere questi passaggi , in quanto il valore atteso di una binomiale è un risultato trovato più volte )

-------------------------
### Calcolo della $Cov(X_1X_2)$

Sappiamo per definizione che:

$$Cov(X_1,X_2)= \mathbb{E}(X_1X_2)-\mathbb{E}(X_1)\mathbb{E}(X_2)$$

Calcoliamo $\mathbb{E}(X_1X_2)$:

$$\mathbb{E}(X_1X_2)=\mathbb{E}(\sum_{i=1}^{n} X_{1i}\sum_{i=1}^{n}X_{2j})$$

Svolgo il prodotto delle due  sommatorie:

$$\mathbb{E}(\sum_{i=j=0}^{n} X_{1i}X_{2j}+\sum_{i\neq j}X_{1i}X_{2j})=\mathbb{E}(\sum_{i=j=0}^{n} X_{1i}X_{2j})+\mathbb{E}(\sum_{i\neq j}X_{1i}X_{2j})=\sum_{i=j=0}^{n}\mathbb{E}( X_{1i}X_{2j})+\sum_{i\neq j}\mathbb{E}(X_{1i}X_{2j})$$

Osservo che :
$$\sum_{i=j=0}^{n} \mathbb{E}(X_{1i}X_{2j})=0$$

Questo perché sappiamo che $X_{1i}$ e $X_{2j}$ (dove $i=j$) sono due variabili aleatorie che stanno contando se l' $i$ -esima pallina è in una delle due scatole. Allora per costruzione delle due variabili aleatorie queste  valgono o $1$ o $0$, di conseguenza quando valgono entrambe $0$  o una delle due vale $0$, allora il  prodotto è nullo .Questo succede anche quando valgono entrambe $1$ , ma in questo caso vale zero $\mathbb{P}(X_{1i}=1 \cap X_{2j}=1)$. Conclusione la sommatoria è a termini tutti nulli e quindi vale $0$.

$$0+\sum_{i\neq j}\mathbb{E}(X_{1i}X_{2j})$$

Tornando all'altra sommatoria , possiamo fare un'altra osservazione che ci semplificherà i calcoli : infatti  in questa somma stiamo considerando solo il valore atteso di due variabili aleatorie indipendenti , questo ci è garantito dal fatto che $i\neq j$.
Quindi gli unici termini non nulli della sommatoria sono quelli in cui sia $X_{1i}$ che $X_{2j}$ valgono $1$, allora il valore atteso vale : 

$$\mathbb{E}(X_{1i}X_{2j})=\frac{1}{k^2}\\qquad\text{per $i\neq j$}$$

Se escludiamo i termini "diagonali" del prodotto tra $X_{1i}$ e $X_{2j}$ (ovvero $n$) , otteniamo $n^2-n$ termini nella seconda sommatoria.

$$\mathbb{E}(X_{1i}X_{2j})=\sum_{i\neq j}\mathbb{E}(X_{1i}X_{2j})=\frac{n^2-n}{k^2}$$

Tornando alla covarianza abbiamo ottenuto che:

$$Cov(X_1,X_2)=\frac{n^2-n}{k^2}-\frac{n^2}{k^2}=-\frac{n}{k^2}$$

**Comment by cornflexxx** [comment]
### Calcolo della distribuzione limite di $X_1$

Per $n,k\to\infty$ abbiamo che $\frac{n}{k}\to\lambda$, partiamo dalla distribuzione iniziale:

$$\mathbb{P}(X_1=x)=\binom{n}{x} (\frac{1}{k})^x (1-\frac{1}{k})^{n-x}$$

Sappiamo già che per $x$ fisso e $n\to\infty$ il coefficiente binomiale si approssima così:
$$\binom{n}{x}\sim\frac{n^x}{x!}$$

Avendo che  $\frac{n}{k}\to\lambda$ allora possiamo dire che $\frac{1}{k}\to\frac{\lambda}{n}$.
Sostituiamo entrambi i risultati:

$$\mathbb{P}(X_1=x)_{n,k\to\infty}=\frac{1}{x!}\cdot (\frac{n}{k})^x\cdot(1-\frac{\lambda}{n})^{n-x}=\frac{\lambda^x}{x!}\cdot e^{-\lambda}$$

Quindi quando il numero di scatole e di biglie diventa infinito la distribuzione binomiale trovata in precedenza diventa una distribuzione di Poisson con $\lambda=np=\frac{n}{k}$.

$$X_1\sim \mathbb{P}(\lambda)\approx Bin(\frac{1}{k},n) \qquad \text{Quando $n,k\to\infty$ }$$

Effettivamente se consideriamo un insieme infinito di palline, e fissiamo con $x$ il numero di palline che vogliamo ottenere , la distribuzione di Poisson  indica proprio la probabilità di ottenere $x$ palline in un determinato intervallo di tempo, in particolare $\lambda$ (in questo caso $\frac{n}{k}$) indica il numero medio delle palline che arrivano in un intervallo temporale scelto.

**Comment by CiottoloMaggico** [comment]
@cornflexxx Nel punto due ho qualche dubbio sul fatto che il valore atteso di X1,X2 sia nullo dato che scelto uno dei due l'altro può assumere valori nel range di [0, n-#palline in scatola 1].

**Comment by cornflexxx** [comment]
> @cornflexxx Nel punto due ho qualche dubbio sul fatto che il valore atteso di X1,X2 sia nullo dato che scelto uno dei due l'altro può assumere valori nel range di [0, n-#palline in scatola 1]. 

In realtà non dico questo , forse hai confuso $X_1$ e $X_2$ con $X_{1i},X_{2j}$ dove $i=j$ , questo significa che indicano la stessa "pallina" che non può stare in entrambe le scatole contemporaneamente.
